doc/lua_api.txt

```diff
@@ -617,6 +640,64 @@ Result is more like what you'd expect if you put a color on top of another
 color, meaning white surfaces get a lot of your new color while black parts
 don't change very much.
 
+A Multiply blend can be applied between two textures by using the overlay 
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:-64^[overlay:textureB.png
+
+#### `[screen:<color>`
+
+Apply a Screen blend with the given color. A Screen blend is the inverse of
+a Multiply blend, lightening images instead of darkening them.
+
+`<color>` is specified as a `ColorString`.
+
+A Screen blend can be applied between two textures by using the overlay 
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:64^[overlay:textureB.png
+
+#### `[hsl:<hue>:<saturation>:<lightness>`
+
+Adjust the hue, saturation, and lightness of the texture. Like
+"Hue-Saturation" in GIMP, but with 0 as the mid-point.
+
+`<hue>` should be from -180 to +180
+
+`<saturation>` and `<lightness>` are optional, and both from -100 to +100
+
+#### `[contrast:<contrast>:<brightness>`
+
+Adjust the brightness and contrast of the texture. Conceptually like
+GIMP's "Brightness-Contrast" feature but allows brightness to be wound 
+all the way up to white or down to black.
+
+`<contrast>` and `<brightness>` are both values from -127 to +127.
```

Comment:
Cast a critical eye at [the 4 lines of code at line 2365](https://github.com/minetest/minetest/pull/10100/files#diff-955e227b6f551c421062f9038477a5f8d4cedf669d96bcefd22e83df5f451df0R2365-R2368)

I can explain my thinking, but that doesn't mean it's best. There are two issues.

I feel that raising brightness and lowering brightness should be symmetric operations, and that wouldn't be the case if it lowers in n/128 fractional intervals but raises in n/127 fractional intervals. However I can't think of any real-world cases where this matters because raising or lowering brightness also lowers contrast, so it's not like you could use it to undo an earlier brightness operation by doing it again with a negative value (unless doing some clever math with the contrast argument).

(The args aren't even signed bytes, the range could be -1000 to +1000, but such accuracy would be misleading with 24bit RGB values. I was probably influences by the UI choices of paint programs and other texture modifiers)

The other issue is how I'm avoiding infinite slope, perhaps a better solution to line 2367 might be
```
double norm_c = core::clamp(contrast,   -127, 127) / 127.0;
if (norm_c == 1.0)
    norm_c = 0.999;

that way a slope of zero is possible (make everything the same grey), but these extremes in a contrast operation don't seem very useful, or visually different from what the existing code does.